### PR TITLE
Add tests switching between fields in AmazonUploadComponent

### PR DESCRIPTION
**Pull request by atodorov**

### CODE REVIEW

File: src/test/SmartComponents/CreateImageWizard/CreateImageWizard.test.js [hunk 1]
```diff
@@ -213,3 +281,41 @@ describe('Step Review', () => {
...
+    test('with valid values', async () => {
+        const next = screen.getByRole('button', { name: /Next/ });
+
+        // select release
+        userEvent.selectOptions(screen.getByTestId('release-select'), [ 'rhel-8' ]);
+        next.click();
+
+        // select upload target
+        await screen.findByTestId('upload-destination');
+        userEvent.selectOptions(screen.getByTestId('upload-destination'), [ 'aws' ]);
```
Comment (atodorov) [inline src/test/SmartComponents/CreateImageWizard/CreateImageWizard.test.js:299]: @Gundersanne this yields `TypeError: Cannot read property 'access_key_id' of null`  and I'm not sure if I'm messing up something or it is a legitimate error, see:
https://travis-ci.com/github/osbuild/image-builder-frontend/builds/200385002. Please advise.

Comment (Gundersanne) [reply src/test/SmartComponents/CreateImageWizard/CreateImageWizard.test.js:299]: hmm, that might be a legitimate error. The wizard should pass state.upload in it's entirety to the uploadcomponent. Which means `setUpload` somehow gets called where it updates the state with an object where `.options` doesn't exist

Comment (atodorov) [reply src/test/SmartComponents/CreateImageWizard/CreateImageWizard.test.js:299]: :beetle: :beetle: :beetle: yay

Comment (atodorov) [reply src/test/SmartComponents/CreateImageWizard/CreateImageWizard.test.js:296]: @Gundersanne IDK if I fixed this issue with my second commit or not but I am seeing a different error now:
`ReferenceError: insights is not defined`, see https://travis-ci.com/github/osbuild/image-builder-frontend/builds/201236611

**Comment by Gundersanne** [comment]
@atodorov i had a look and made a commit here: https://github.com/Gundersanne/image-builder-frontend/tree/51-test

- `npm run test:single` runs jest with `-w 1`, makes the output readable
- adds testIds to the radio buttons, I thnk it's a bit bugged where if you click the label it'll automatically switch to the first radio button (even if you click the label of the 2nd button)
- moved a misplaced wait

When running in parallel I still get this warning :
```
console.error
    Warning: You seem to have overlapping act() calls, this is not supported. Be sure to await previous act() calls before making a new one.
```
It doesn't make the tests fail, but sadly I have no idea what that's about.

**Comment by atodorov** [comment]
> When running in parallel I still get this warning :
> 
> ```
> console.error
>     Warning: You seem to have overlapping act() calls, this is not supported. Be sure to await previous act() calls before making a new one.
> ```
> 
> It doesn't make the tests fail, but sadly I have no idea what that's about.

@Gundersanne this is an internal function from the testing library, see
https://testing-library.com/docs/preact-testing-library/api#act. IDK why it gets reported but if it doesn't fail then don't fix it, right ?

**Comment by atodorov** [comment]
> @atodorov i had a look and made a commit here: [Gundersanne@17a69cb](https://github.com/Gundersanne/image-builder-frontend/commit/17a69cb037a649b9f55c560e41255359d278e5d0)

@Gundersanne thanks. I've cherri-picked your commit and it seems to work. Everything is green so I say let's merge this PR.

**Comment by Gundersanne** [comment]
@atodorov sorry about this. I forgot to merge this before renaming the master branch to main Could you reopen this against the main branch?

**Pull request closed without merging**